pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -444,3 +515,50 @@ func (policy *regoEnforcer) ExtendDefaultMounts(mounts []oci.Mount) error {
 func (policy *regoEnforcer) EncodedSecurityPolicy() string {
 	return policy.base64policy
 }
+
+func (policy *regoEnforcer) EnforceExecInContainerPolicy(containerID string, argList []string, envList []string, workingDir string) error {
+	policy.mutex.Lock()
+	defer policy.mutex.Unlock()
+
+	input := map[string]interface{}{
+		"containerID": containerID,
+		"argList":     argList,
+		"envList":     envList,
+		"workingDir":  workingDir,
+	}
+
+	results, err := policy.query("exec_in_container", input)
+	if err != nil {
+		return err
+	}
+
+	allowed, err := policy.allowed("exec_in_container", results)
+	if err != nil {
+		return err
+	}
+
+	if allowed {
+		err = policy.updateMetadata(results)
+		if err != nil {
+			return fmt.Errorf("unable to update metadata: %w", err)
+		}
+		return nil
+	} else {
+		input_json, err := json.Marshal(input)
+		if err != nil {
+			return fmt.Errorf("unable to marshal the Rego input data: %w", err)
+		}
+
+		input["rule"] = "exec_in_container"
+		results, err := policy.query("reason", input)
+		if err != nil {
+			return fmt.Errorf("exec in container not allowed by policy.\nInput: %s", string(input_json))
```

Comment:
don't put `\n` into the error message here and in other places too. We agreed that we'll take care of already existing error messages like this, but moving forward lets avoid this pattern.
If we don't want to return the error here, can we at least log it?

Reply:
What would your preferred pattern be here? Happy to comply with accepted code style, just let me know what this (and other errors like it) should look like.